### PR TITLE
Feature/limit favors per user

### LINKED ISSUE #10: As a user who already accepted a favor, I want the system to make me complete or leave the current favor before accepting another favor at the same time, so that I don't need to worry about how to coordinate multiple accepted favors

### LINKED ISSUE #15: As a user who has already posted a favor, I want to be able to post other favors, so that I can receive help for different situations and times. I will be able to ask up to N favors at a time.

### LINKED ISSUE #172: When I log in for the first time I'm unable to receive notifications

### LINKED ISSUE #182: As a user, I'm annoyed that I'm seeing pictures in Favors that should not have Favors.
When a Favor is clicked on in the active/archived, the observer in FavorRequestView runs the whole display Favor code twice, once with the old Favor and once with the new Favor. This causes pictures to show up in Favors that do not have pictures.

This happens from lines 109 to 122 in `FavorRequestView`. We suspect it has something to do with the `LiveData` class.

picture_bug

### PR DESCRIPTION
**Pull request by danielpgp1012**
Fixes sign in bug  (closes #172)
Fixes listener duplication (closes #182)
Implements max requested and accepted favors flow (closes #10, closes #15)

### CODE REVIEW

File: app/src/main/java/ch/epfl/favo/user/UserUtil.java [hunk 1]
```diff
@@ -95,22 +116,32 @@ public void deleteAccount() {
...
+  //              if (!task.isSuccessful()) {
+  //                return;
+  //              }
+  //              String token = Objects.requireNonNull(task.getResult()).getToken();
+  //              user.setNotificationId(token);
+  //
+  //              Map<String, String> notifMap = new HashMap<String, String>();
+  //              notifMap.put("notificationId", token);
+  //              collection.updateDocument(user.getId(), notifMap);
+  //            });
```
Comment (arosu) [inline app/src/main/java/ch/epfl/favo/user/UserUtil.java:144]: Is there a reason this is commented out but not deleted?

Comment (danielpgp1012) [reply app/src/main/java/ch/epfl/favo/user/UserUtil.java:144]: Yeah you're right I forgot to delete it

File: app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java [hunk 2]
```diff
@@ -74,14 +79,14 @@ public FavorRequestView() {
   @Override
   public View onCreateView(
       LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-
     View rootView = inflater.inflate(R.layout.fragment_favor_request_view, container, false);
+    // getActivity().setContentView(R.layout.fragment_favor_request_view);
```
Comment (arosu) [inline app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java:83]: Same as before.

File: app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java [hunk 3]
```diff
@@ -217,28 +230,31 @@ private void toggleTextViewsEditable(boolean switchToEditable) {
   private void requestFavor() {
     // update currentFavor
     View currentView = getView();
-    favorStatus = FavorStatus.REQUESTED;
-    getFavorFromView(favorStatus);
-
+    // enableUploadImageButtons(false);
+    // confirmFavorBtn.setEnabled(false);
```
Comment (arosu) [inline app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java:234]: Same thing here.

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -22,7 +22,7 @@
         android:roundIcon="@drawable/logo"
         android:supportsRtl="true"
         android:theme="@style/AppTheme.Launcher"
-        android:windowSoftInputMode="adjustNothing">
+        android:windowSoftInputMode="adjustNothing|adjustPan">
```
Comment (mikanikos) [inline app/src/main/AndroidManifest.xml:25]: Why adding adjustPan? Maybe adjustResize would be better but is there a reason to add it?

Comment (danielpgp1012) [reply app/src/main/AndroidManifest.xml:25]: Yeah so for some reason the keyboard was moving up the buttons again. I just realized that the chat view is also moving up the toolbar and the message on top. I will try to fix it

Comment (mikanikos) [reply app/src/main/AndroidManifest.xml:25]: Yeah indeed I changed it to adjustNothing in the chatPage PR to make the messages show correctly while typing and didn't notice the buttons issue, my bad. Maybe there's a way to separate the behavior of the soft keyboard in different pages.

**Pull request merged**